request.go

```diff
@@ -0,0 +1,88 @@
+package cfdns
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"golang.org/x/exp/maps"
+)
+
+type request[T any] struct {
+	client      *Client
+	method      string
+	path        string
+	queryParams url.Values
+	headers     http.Header
+	body        *T
```

Comment:
This should be an pointer to pointer, because any is an interface which is a pointer, is that on purpose?

```suggestion
	body        T
```

Reply:
If the request is `nil` I don't marshal it on the request body, and I can't do `if body == nil {}` if this line is `body T`. If I change the request body to be `any`, as you suggested, then this will no longer be necessary.

Reply:
Issue solved by using `any` for body instead of a type parameter.